"Resource not found" error if running with old Gradle < 1.4

With a `build.gradle` containing only the following:

```
plugins {
  id "com.github.spotbugs" version "1.3"
}
```

(i.e., the build-script snippet from https://plugins.gradle.org/plugin/com.github.spotbugs), I get the following output from `gradlew.bat --info`:

```
Starting Build
Settings evaluated using settings file 'C:\Users\m.gould\Documents\GitHub\ilab-ww\settings.gradle'.
Projects loaded. Root project using build file 'C:\Users\m.gould\Documents\GitHub\ilab-ww\build.gradle'.
Included projects: [root project 'ai']
Evaluating root project 'ai' using build file 'C:\Users\m.gould\Documents\GitHub\ilab-ww\build.gradle'.

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\m.gould\Documents\GitHub\ilab-ww\build.gradle' line: 2

* What went wrong:
An exception occurred applying plugin request [id: 'com.github.spotbugs', version: '1.3']
> Failed to apply plugin [id 'com.github.spotbugs']
   > resource spotbugs-gradle-plugin.properties not found.

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output.
```

The output of `gradlew.bat --version` is as follows:

```
------------------------------------------------------------
Gradle 2.9
------------------------------------------------------------

Build time:   2015-11-17 07:02:17 UTC
Build number: none
Revision:     b463d7980c40d44c4657dc80025275b84a29e31f

Groovy:       2.4.4
Ant:          Apache Ant(TM) version 1.9.3 compiled on December 23 2013
JVM:          1.8.0_131 (Oracle Corporation 25.131-b11)
OS:           Windows 10 10.0 amd64
```

Things I have tried:

 - Installing version 1.2 instead (same error)
 - Creating files called `spotbugs-gradle-plugin.properties` in my project's root directory, and under every `resources` directory I could find (same error)
 - Using the pre-2.1 build-script syntax given on that page (different error! "Could not generate a proxy class for class com.github.spotbugs.SpotBugsTask.")
 - Running `jar tf` on the downloaded `gradlePlugin.jar` - the file `spotbugs-gradle-plugin.properties` is indeed present, at the root.

What am I doing wrong?

--- a/gradlePlugin/CHANGELOG.md
+++ b/gradlePlugin/CHANGELOG.md
@@ -6,6 +6,8 @@ Currently the versioning policy of this project does not follow [Semantic Versio
 
 ## Unreleased
 
+* Make build failed when user uses unsupported Gradle version ([#357](https://github.com/spotbugs/spotbugs/issues/357))
+
 ## 1.4 - 2017-09-25
 
 * Use SpotBugs 3.1.0-RC6

--- a/gradlePlugin/src/main/java/com/github/spotbugs/SpotBugsPlugin.java
+++ b/gradlePlugin/src/main/java/com/github/spotbugs/SpotBugsPlugin.java
@@ -24,6 +24,7 @@ import org.gradle.api.plugins.quality.internal.AbstractCodeQualityPlugin;
 import org.gradle.api.reporting.SingleFileReport;
 import org.gradle.api.resources.TextResource;
 import org.gradle.api.tasks.SourceSet;
+import org.gradle.util.GradleVersion;
 
 /**
  * A plugin for the <a href="https://spotbugs.github.io">SpotBugs</a> byte code analyzer.
@@ -41,6 +42,12 @@ import org.gradle.api.tasks.SourceSet;
  */
 public class SpotBugsPlugin extends AbstractCodeQualityPlugin<SpotBugsTask> {
 
+    /**
+     * Supported Gradle version described at <a href="http://spotbugs.readthedocs.io/en/latest/gradle.html">official
+     * manual site</a>.
+     */
+    private static final GradleVersion SUPPORTED_VERSION = GradleVersion.version("4.0");
+
     private SpotBugsExtension extension;
 
     @Override
@@ -55,7 +62,24 @@ public class SpotBugsPlugin extends AbstractCodeQualityPlugin<SpotBugsTask> {
 
     @Override
     protected void beforeApply() {
+        verifyGradleVersion(GradleVersion.current());
         configureSpotBugsConfigurations();
+    }
+
+    /**
+     * Verify that given version is supported by {@link SpotBugsPlugin} or not.
+     * 
+     * @param version
+     *            to verify
+     * @throws IllegalArgumentException
+     *             if given version is not supported
+     */
+    void verifyGradleVersion(GradleVersion version) throws IllegalArgumentException {
+        if (version.compareTo(SUPPORTED_VERSION) < 0) {
+            String message = String.format("Gradle version %s is unsupported. Please use %s or later.", version,
+                    SUPPORTED_VERSION);
+            throw new IllegalArgumentException(message);
+        }
     }
 
     private void configureSpotBugsConfigurations() {

--- a/gradlePlugin/src/test/java/com/github/spotbugs/SpotBugsPluginTest.java
+++ b/gradlePlugin/src/test/java/com/github/spotbugs/SpotBugsPluginTest.java
@@ -16,6 +16,7 @@ import org.gradle.testkit.runner.BuildResult;
 import org.gradle.testkit.runner.BuildTask;
 import org.gradle.testkit.runner.GradleRunner;
 import org.gradle.testkit.runner.TaskOutcome;
+import org.gradle.util.GradleVersion;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Rule;
@@ -96,4 +97,34 @@ public class SpotBugsPluginTest extends Assert{
     SpotBugsPlugin plugin = new SpotBugsPlugin();
     assertThat(plugin.loadToolVersion(), is(notNullValue()));
   }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testVersionVerifyForGradleVersion2() {
+        SpotBugsPlugin plugin = new SpotBugsPlugin();
+        plugin.verifyGradleVersion(GradleVersion.version("2.0"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testVersionVerifyForGradleVersion3() {
+        SpotBugsPlugin plugin = new SpotBugsPlugin();
+        plugin.verifyGradleVersion(GradleVersion.version("3.0"));
+    }
+
+    @Test
+    public void testVersionVerifyForGradleVersion4() {
+        SpotBugsPlugin plugin = new SpotBugsPlugin();
+        plugin.verifyGradleVersion(GradleVersion.version("4.0"));
+    }
+
+    @Test
+    public void testVersionVerifyForGradleVersion41() {
+        SpotBugsPlugin plugin = new SpotBugsPlugin();
+        plugin.verifyGradleVersion(GradleVersion.version("4.1"));
+    }
+
+    @Test
+    public void testVersionVerifyForGradleVersion42() {
+        SpotBugsPlugin plugin = new SpotBugsPlugin();
+        plugin.verifyGradleVersion(GradleVersion.version("4.2"));
+    }
 }

